### ISSUE TITLE
Undefined method on an instance of the wrong class

### ISSUE THREAD

**Issue by wayneashleyberry** [YorickPeterse/ruby-lint#199]
I'm getting an error message for the wrong class. The first error message that ruby-lint is reporting is on line 15 but that's actually inside `class Bar` not `class Baz`.

```
$ /usr/local/lib/ruby/gems/2.3.0/gems/ruby-lint-2.3.1/bin/ruby-lint test.rb
foo.rb: error: line 15, column 5: undefined method surname= on an instance of Baz
foo.rb: warning: line 20, column 5: unused instance variable @surname
foo.rb: warning: line 32, column 5: unused instance variable @middle_name
foo.rb: warning: line 36, column 1: unused local variable bar
foo.rb: warning: line 37, column 1: unused local variable baz
```

```ruby
class Foo
  def initialize(name = 'Anonymous')
    self.name = name
  end

  def name=(name)
    raise 'Name cannot be empty!' if name.empty?
    @name = name
  end
end

class Bar < Foo
  def initialize(name = 'Anonymous', surname = 'Bear')
    super name
    self.surname = surname # <~ The error is being reported here!
  end

  def surname=(surname) # <~ The method clearly exists on this class.
    raise 'Surname cannot be empty!' if surname.empty?
    @surname = surname
  end
end

class Baz < Foo
  def initialize(name = 'Anonymous', middle_name = 'Fox')
    super name
    self.middle_name = middle_name
  end

  def middle_name=(middle_name)
    raise 'Middle name cannot be empty!' if middle_name.empty?
    @middle_name = middle_name
  end
end

bar = Bar.new('lk1j2lk')
baz = Baz.new('asdjkfl')
```

**Comment by mvidner** [comment]
Confirmed. And here is a smaller test case:
```console
$ ruby-lint i199.rb 
i199.rb: error: line 6, column 5: undefined method bar on an instance of Baz
```
i199.rb:
```rb
class Foo
end

class Bar < Foo
  def initialize
    self.bar
  end

  def bar
  end
end

class Baz < Foo
  def initialize
    self.baz
  end

  def baz
  end
end
```
